lib/Transforms/Scalar/PromoteVerifierCalls.cc

```diff
@@ -32,6 +34,41 @@ Value *coerceToBool(Value *arg, IRBuilder<> &builder, Instruction &I) {
     arg = builder.CreateICmpNE(arg, ConstantInt::get(arg->getType(), 0));
   return arg;
 }
+
+/// Determines if \p v represents a partial function call. If so, then the
+/// function call is returned.
+Value *extractPartialFnCall(Value *v) {
+  // The partial call may be cast to bool by comparison to 0.
+  if (auto cmp = dyn_cast<ICmpInst>(v)) {
+    auto *lhs = cmp->getOperand(0);
+    auto *rhs = cmp->getOperand(1);
+    v = isa<CallInst>(lhs) ? lhs : rhs;
+  }
+
+  // The value must now be a call instruction.
+  if (!isa<CallInst>(v))
+    return nullptr;
+
+  // Annotation2Metadata marks all instructions of the partial function with the
+  // "partial" annotation. Partial functions must have at least one instruction
+  // (i.e., a return) so this check is robust.
+  CallSite CS(v);
```

Comment:
callsite is deprecated and removed in llvm11. should be a way to use callbase instead